### DISCUSSION TITLE
Reset Client ID and Secret

### DISCUSSION THREAD

**Discussion by smalldatach** [astrandb/miele#247] [Q&A] [answered]
Hi

How can I reset / delete current Client ID and Secret ID?
Uninstall / Reboot / Install / Rebot does not help. Everytime the wizard starts, it goes directly to the website from miele...

Thanks!!!

**Comment by astrandb** [comment, marked as answer]
Goto Settings->Integrations and open the overflow menu (three vertical dots) and you will find what you are looking for.

**Reply by smalldatach** [reply]
perfect awnser!!! Thanks :)

**Reply by Epimedium** [reply]
Hi,
I have the same issue. The only thing is that I deleted my integration and now it is not in my integration overview anymore. Whenever I reinstall via HACS and then try to add the integration again, it uses my old credentials (which I thought I deleted). There are no "three dots" available at any stage. I'm prompted to go to the Miele website directly without the option to fill in credentials.
I hope you have a solution for this.

**Reply by astrandb** [reply]
![image](https://github.com/astrandb/miele/assets/5016010/1b18f08f-7e8d-483a-8249-17766ba0796e)

**Reply by Epimedium** [reply]
Many thanks!

**Comment by almiho** [comment]
hi all,

I have the same issue, I need to enter new client ID and secret but when I hit the "configure" button on the integrations page I am taken directly to a Miele page which tells me that my client ID and secret are not valid. That's why I want to update them but the Miele integration won't let me. 

I have installed version v0.1.16 of the Miele integration on HA 2023.10.3.

I have uninstalled the Miele integration and rebooted a few times but no luck. The integration shows up on the "integrations" page.

<img width="330" alt="Screenshot 2023-10-18 at 11 38 02" src="https://github.com/astrandb/miele/assets/17482115/47ccc1b7-fc6d-440b-8a8b-a44599e6c824">



when I hit the configure button I get forwarded directly to the error page of Miele complaining about invalid credentials
<img width="694" alt="Screenshot 2023-10-18 at 11 39 24" src="https://github.com/astrandb/miele/assets/17482115/b8046a7b-6577-4e67-bbd2-a9c91226d0a7">




when I click the three dots as suggested in an earlier repsonse the only option I have is to go to the documentation page on github
<img width="476" alt="Screenshot 2023-10-18 at 11 40 09" src="https://github.com/astrandb/miele/assets/17482115/506640bb-54e1-4dbc-bf3b-d9bf1c6f2999">



when I try to re-add the Miele Integration I get to the following dialog which claims that I have to enter my credentials.
<img width="653" alt="Screenshot 2023-10-18 at 11 43 45" src="https://github.com/astrandb/miele/assets/17482115/f18a38c3-c0c3-409c-a5ba-091141daf8ae">


unfortunately what the previous dialog says is not true. I get no chance to enter my credentials, I just get back to the error page from Miele with the outdated old credentials.
<img width="661" alt="Screenshot 2023-10-18 at 11 44 40" src="https://github.com/astrandb/miele/assets/17482115/c729b4b9-c8ed-410c-8cd9-660b40413be9">

Can someone help me understand where these old credentials are stored and how I delete them?

Thanks,
Alex

**Reply by astrandb** [reply]
> Can someone help me understand where these old credentials are stored and how I delete them?

Please see my answer above.

**Comment by almiho** [comment]
ok, you were right, I just didn't read it carefully enough.

For everyone who has the same issues like me, here are the two important things that helped me:

- click the three dots on the HA integrations page (NOT the three dots on the Miele integration)
- after deleting the credentials via the page "hidden" behind the three dots of the integration page (!) make sure to reboot HA so that the credentials actually get deleted

**Reply by metawops** [reply]
Okay, thanks for your hints, @almiho.
This is what happens for me here, after I removed the credentials, restarted HA, started the process again, entered my (newly created, so 2nd set of) credentials (client ID, secret): I enter the Name and both credentials and – nothing happens. The dialog disappears and that's it. No next step with a button that will open a miele.com page. Nothing.
🤷‍♂️ 
I'm on macOS and my browser is Safari, btw.
And I removed all browser data from domain miele.com, too.

The entered credentials are saved in HA, I can see the new entry via the three dots menu.

**Reply by CNjord** [reply]
Did you find out what was wrong I have the same problem?

**Reply by puetsch** [reply]
Thanks @almiho this was super helpful! I was completely stuck but this helped me reset the credentials.

@astrandb I think this could be a worthwhile addition to the wiki. What do you think? Let me know if I can help.
